src/lib.rs

```diff
@@ -258,4 +295,14 @@ mod tests {
         })
         .unwrap();
     }
+
+    #[test]
+    #[should_panic]
+    fn exec_function_assert_test() {
+        exec_function(|| {
+            assert_eq!(true, false);
+            Ok(())
+        })
+        .unwrap();
```

Comment:
It may be better to check if you can get mutex after assertion is failed.

Reply:
@bluele Thanks. updated.